src/osd/PGLog.h

```diff
@@ -1239,8 +1237,8 @@ struct PGLog : DoutPrefixProvider {
     eversion_t dirty_to,
     eversion_t dirty_from,
     eversion_t writeout_from,
-    const set<eversion_t> &trimmed,
-    const set<string> &trimmed_dups,
+    set<eversion_t> *trimmed,
```

Comment:
could pass by rvalue reference, which reflects the intention of this function better. the up side of using pointer is that it saves the overhead of constructing a temporary empty `set<>`,  but the only case we pass `nullptr` to `_write_log_and_missing()` is in `ceph_objectstore_tool.cc`, so I don't think that's a bottleneck we need to worry about.

Reply:
fixed